spec/factories/courses.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :course do
+    provider
+    sequence(:name) { |c| "Course #{c}" }
+    code { Faker::Alphanumeric.alphanumeric(number: 3).upcase }
```

Comment:
```suggestion
    code { Faker::Alphanumeric.alphanumeric(number: 4).upcase }
```
![image](https://user-images.githubusercontent.com/470137/110959651-d2f22a00-8345-11eb-896f-d3357c0a153f.png)